### DISCUSSION TITLE
Add option to ignore extra fields in config

### DISCUSSION THREAD

**Discussion by okainov** [renovatebot/renovate#24918] [Suggest an Idea]
### What would you like Renovate to be able to do?

To parse `renovate.json` with extra fields successfully.

Currently we're getting error:

> Error type: The renovate configuration file contains some invalid settings
> Message: Invalid configuration option: myCustomOption

We run some of our extra scripts doing basically the similar activities as Renovate but for scenarios which are not supported by Renovate and/or too specific for our team to be implemented in public Renovate. We find it convenient to put the options directly into `renovate.json` to do not create multiple files in the repo. However now we get an error even though the rest of the config is still perfectly valid...

### If you have any ideas on how this should be implemented, please tell us here.

Add an option like `--ignore-extra-fields` to Renovate run or so

### Is this a feature you are interested in implementing yourself?

No

**Comment by rarkins** [comment]
I think we could maybe accept an option to skip config validation completely, but it would need to be a community contribution

**Comment by rarkins** [comment]
@viceice @JamieMagee we need to decide if we want to skip config validation completely, or allow extra fields, or both

**Comment by viceice** [comment]
I don't really like any of them. but I'm ok with both.

**Comment by viceice** [comment]
I think this should also only be possible for config files. so we don't need the bad import from config to worker in PR.

**Comment by rarkins** [comment]
Sorry, can you clarify the last point?

**Comment by viceice** [comment]
I think we don't need to change anything for cli and env vars, they should be validated as is now. So those extra values can only be occure in config files, so only those need to be skiped ...

**Comment by rarkins** [comment]
Good point, agreed

**Comment by rarkins** [comment]
I'm wondering if this should instead be a RENOVATE_X_SKIP_CONFIG_VALIDATION option because its use should be very rare

**Comment by viceice** [comment]
> I'm wondering if this should instead be a RENOVATE_X_SKIP_CONFIG_VALIDATION option because its use should be very rare

agreed 👍
